x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -80,6 +80,10 @@ public boolean allowSystemIndexAccessByDefault() {
         return restHandler.allowSystemIndexAccessByDefault();
     }
 
+    public RestHandler getRootRestHandler() {
+        return restHandler;
```

Comment:
I think this should be 
```suggestion
        return restHandler.getRootRestHandler();
```
in case we have Security wrapper around a Deprecation handler.